openeo_driver/ProcessGraphDeserializer.py

```diff
@@ -1454,6 +1454,10 @@ def run_udf(args: dict, env: EvalEnv):
         dataframe = geo_data
         if isinstance(geo_data, gpd.GeoSeries):
             dataframe = gpd.GeoDataFrame(geometry=geo_data)
+        invalid_indexes = dataframe.index[~dataframe.is_valid].tolist()
+        if len(invalid_indexes) > 0:
+            raise ValueError("UDF returned invalid polygons. This could "
```

Comment:
If you raise an exception with a message for the user to see, you should raise a `OpenEOApiException` (preferably one of its subclasses, if applicable), otherwise the user will receive a generic "500 Internal server error"

In this case it's a user error, so it should at least be a status 400. There is not really a predefined error code, but you can invent one and specify that:

```suggestion
            raise OpenEOApiException(
                status_code=400,
                code="InvalidGeometry",
                message="UDF produced invalid geometry. This could "
```

Reply:
I applied this and added a warning to the driver vector cube constructor. I'll merge to master already to see how the tests go

Reply:
No `DriverVectorCube with invalid polygon` messages in python-driver nor geopyspark-driver nor integration test logs found. But still I don't dare to throw an actual exception. Maybe after checking the kibana logs in a few weeks